mne/morph.py

```diff
@@ -0,0 +1,953 @@
+                              VectorSourceEstimate)
+from .source_space import SourceSpaces
+from .surface import read_surface, read_morph_map, mesh_edges
+from .utils import (logger, verbose, get_subjects_dir, warn as warn_,
+                    has_nibabel, has_dipy)
+
+
+class SourceMorph(object):
+    """Container for source estimate morphs.
+
+    Parameters
+    ----------
+    src : instance of SourceSpaces
+        Information about the respective source space, with src being the
+        corresponding list of SourceSpaces
+    subject_from : str | None
+        Name of the original subject as named in the SUBJECTS_DIR.
+        If None src[0]['subject_his_id]' will be used. The default is None.
+    subject_to : str
+        Name of the subject on which to morph as named in the SUBJECTS_DIR
+        The default is 'fsaverage'.
+    data_from : array | list of two arrays | str | None
+        If morphing a surface source estimate, data_from is the
+        Vertex numbers corresponding to the data.
+        If morphing a volume source estimate data_from is the absolute path
+        where subject_from/mri/brain.mgz is stored
+        If None data_from will be found through subject_from if present. The
+        default is None.
+    data_to : array | list of two arrays | str | None
+        If morphing a surface source estimate , data_to is the
+        Vertex numbers corresponding to the data.
+        If morphing a volume source estimate data_to is the absolute path
+        where subject_to/mri/brain.mgz is stored
+        If None data_to will be found through subject_to if present. The
+        default is None.
+    subjects_dir : str | None
+        Path to SUBJECTS_DIR if it is not set in the environment. The default
+        is None.
+    **options : **kwargs
+        Keyword arguments for the respective morphing function.
+
+    Attributes
+    ----------
+    kind : str | None
+        Kind of source estimate. E.g. 'volume' or 'surface'
+    subjects_dir : str | None
+        The path to the FreeSurfer subjects reconstructions.
+        It corresponds to FreeSurfer environment variable SUBJECTS_DIR.
+    subject_from : str | None
+        Name of the subject from which to morph as named in the SUBJECTS_DIR
+    subject_to : str
+        Name of the subject to which to morph as named in the SUBJECTS_DIR
+    vol_info : dict
+        Information about reference volumes for volumetric morph. Additional
+        information required to transform volume source estimate into
+        Nifti1Image using as_volume
+    src_info : dict
+        Information required to transform volume source estimate into
+        Nifti1Image using as_volume
+    morph_data : Trans | dict
+        If morphing a surface source estimate , morph_data is an instance
+        of Trans to morph the data
+        If morphing a volume source estimate morph_data is a dict storing
+        linear and non-linear mapping information.
+    options : **kwargs
```

Comment:
wait, why is this also an attribute? That's an impressive number of attributes. We should be mindful what we expose as public API.

Reply:
In this case it's because the user could wonder, what the morph was, that was performed. So I think it's nice to have it somehow "logged"

Reply:
we try to avoid kwargs as much as possible as it's hard to document. You see here that by just by reading the doc I don't know what options can contain? can you list these options? if it's 2-3 maybe we actually add them explicitly

Reply:
totally agree with @agramfort 

@TommyClausner for what you mention you need a repr method to your class

Reply:
but @jasmainak then a pseudo-repr as in the other classes, correct? (not a repr that could actually be evaluated)